nautobot/core/views/__init__.py

```diff
@@ -220,3 +225,34 @@ def render_graphiql(self, request, **data):
         data["saved_graphiql_queries"] = GraphQLQuery.objects.all()
         data["form"] = GraphQLQueryForm
         return render(request, self.graphiql_template, data)
+
+
+class NautobotAppMetricsCollector(Collector):
+    """Custom Nautobot metrics collector.
+
+    Metric collector that reads from registry["plugin_metrics"] and yields any metrics registered there."""
+
+    def collect(self):
+        """Collect metrics from plugins."""
+        start = time.time()
+        for metric in registry["app_metrics"]:
+            yield from metric()
+        gauge = GaugeMetricFamily("nautobot_app_metrics_processing_ms", "Time in ms to generate the app metrics")
+        duration = time.time() - start
+        gauge.add_metric([], format(duration * 1000, ".5f"))
+        yield gauge
+
+
+def custom_app_metric_view(request):
```

Comment:
Let's see if we can subclass the existing Django Prometheus view, `ExportToDjangoView`.

Reply:
Unfortunately this is not possible. I have added a comment that explains this into the code.